docs/azure/sdk/protocol-convenience-methods.md

```diff
@@ -0,0 +1,132 @@
+ms.topic: conceptual
+ms.custom: devx-track-dotnet, engagement-fy23, devx-track-arm-template
+ms.date: 06/24/2024
+---
+
+# Azure SDK for .NET protocol and convenience methods overview
+
+The Azure SDK client libraries provide an interface to Azure service REST APIs by translating method calls into messages sent via the respective service protocol. For REST API services, this means sending HTTP requests and converting the responses into runtime types. In this article, you'll learn about the different types of methods exposed by the client libraries and explore their implementation patterns.
+
+## Understand protocol and convenience methods
+
+The Azure SDK for .NET client libraries can expose two different categories of methods to make requests to an Azure service:
+
+- **Protocol methods** provide a thin wrapper around the underlying REST API for a corresponding Azure service. These methods map primitive input parameters to HTTP request values and return a raw HTTP response object.
+
+- **Convenience methods** provide a convenience layer over the lower-level protocol layer to add support for the .NET type system and other benefits. Convenience methods accept primitives or .NET model types as parameters and map them to the body of an underlying REST API request. These methods also handle various details of request and response management to allow developers to focus on sending and receiving data objects, instead of lower-level concerns.
+
+### Azure SDK client library dependency patterns
+
+Protocol and convenience methods implement slightly different patterns based on the underlying package dependency chain of the respective library. The Azure SDK for .NET client libraries depend on one of two different core libraries:
+
+- [**Azure.Core**](/dotnet/api/overview/azure/core-readme) provides shared primitives, abstractions, and helpers for building modern Azure SDK client libraries. These libraries follow the [Azure SDK Design Guidelines for .NET](https://azure.github.io/azure-sdk/dotnet_introduction.html) and use package names and namespaces prefixed with *Azure*, such as [`Azure.Storage.Blobs`](/dotnet/api/overview/azure/storage.blobs-readme).
+
+- [**System.ClientModel**](/dotnet/api/overview/azure/system.clientmodel-readme) is a core library that provides shared primitives, abstractions, and helpers for .NET service client libraries. The `System.ClientModel` library is a general purpose toolset designed to help build libraries for a variety of platforms and services, whereas the `Azure.Core` library is specifically designed for building Azure client libraries.
+
+> [!NOTE]
+> The `Azure.Core` library itself also depends on `System.ClientModel` for various service building blocks. In the context of this article, the key differentiator for method patterns is whether a client library depends on `Azure.Core` or `System.ClientModel` directly, rather than through a transitive dependency.
```

Comment:
> In the context of this article, the key differentiator for method patterns is whether a client library depends on `Azure.Core` or `System.ClientModel` directly, rather than through a transitive dependency.

I believe Azure.AI.OpenAI doesn't have a direct dependency on SCM, but it is an SCM-based client.  It might be worth differentiating differently.  The simplest way might be to look at whether the client constructor takes ClientOptions that inherit from [Azure.Core.ClientOptions](https://learn.microsoft.com/en-us/dotnet/api/azure.core.clientoptions?view=azure-dotnet) or [System.ClientModel.Primitives.ClientPipelineOptions](https://learn.microsoft.com/en-us/dotnet/api/system.clientmodel.primitives.clientpipelineoptions?view=azure-dotnet-preview).  Although this may be a bit complex.

Another approach is to look at whether convenience methods return [Azure.Response\<T\>](https://learn.microsoft.com/en-us/dotnet/api/azure.response-1?view=azure-dotnet) or [System.ClientModel.ClientResult\<T\>](https://learn.microsoft.com/en-us/dotnet/api/system.clientmodel.clientresult-1?view=azure-dotnet-preview). Alternatively, users could check whether protocol methods return [Azure.Response](https://learn.microsoft.com/en-us/dotnet/api/azure.response?view=azure-dotnet) or [System.ClientModel.ClientResult](https://learn.microsoft.com/en-us/dotnet/api/system.clientmodel.clientresult?view=azure-dotnet-preview)

Reply:
@annelo-msft I'm trying to think of how to provide a succinct definition that is more conceptual rather than based on types in code examples. Should we just say something along the lines of:

"In the context of this article, the key differentiator for method patterns is whether a client library depends on `Azure.Core` or `System.ClientModel` directly, rather than through a transitive dependency. However, there are some exceptions to this rule, such as Azure.AI.OpenAI, which follows SCM patterns but depend on Azure.Core to gain access to certain types."